users.py

```diff
@@ -11,7 +11,7 @@ def __init__(self, email, name, contact_number, listings):
     def from_firestore(email, source):
         return User(email, source['profile']['name'], source['profile']['contact_number'], source['listings'])
 
-    def to_firestore(self):
+    def to_firestore(listing_id, self):
```

Comment:
this won't work, as the first argument in a class method is `self` representing the `User` object itself

Reply:
oh right, my bad, good spot! thanks!